### ISSUE TITLE
v8.0.0-beta.1 crashes on start-up when not sandboxed (at least on Linux)

### ISSUE THREAD

**Issue by jacobq** [electron/electron#20856]
### Preflight Checklist

* [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [X] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * v8.0.0-beta.1
* **Operating System:**
  * Linux 5.2.0-3-amd64 # 1 SMP Debian 5.2.17-1 (2019-09-26) x86_64 GNU/Linux
* **Last Known Working Electron version:**
  * 7.0.0

`node -v` --> `v12.13.0` but also tried on latest 8.x and 10.x and had the same results.

### Expected Behavior

`electron-quick-start` / "Hello world" application should run (and not crash).

### Actual Behavior

Main content appears blank / white. Opening Dev Tools results in "DevTools was disconnected from the page...." (reloading only re-triggers the problem)

```
./node_modules/electron/dist/electron --type=renderer ...\
  ../../third_party/electron_node/src/env.cc:256:void node::Environment::CreateProperties(): Assertion `primordials->IsObject()' failed.
```

### To Reproduce

Run the default code in electron-fiddle or electron-quick-start against electron v8.0.0-beta.1 (on Linux -- not sure about other OS). e.g.
https://gist.github.com/jacobq/2e3dcbb82bb9b25287d9ddc513c756b1

### Screenshots

![image](https://user-images.githubusercontent.com/1633459/67880434-9fc24880-fb0c-11e9-816c-2fc719cdb49b.png)

**Comment by vladimiry** [comment]
I remember I experienced a similar issue when I was initially trying the v8 beta but I didn't place the issue here. Can you try explicitly enabling the sandbox flag in webPreferences and see if there is a difference:
```
    webPreferences: {
      ...
      sandbox: true,
      ...
    }
```

**Comment by jacobq** [comment]
@vladimiry thanks for the tip! There certainly is a difference. Setting `sandbox: true` resolved the problem (app starts and shows version info as expected). Setting `sandbox: false` (or omitted) results in the failure described above.

Would you mind providing some explanation / links as to what is going on here?

**Comment by vladimiry** [comment]
What I noticed running the v8-beta1 is that it starts with --no-sandbox option in dev mode (unpackaged app), at least the quick-start-app. It was not like that in the previous electron versions. This is why I went with enabling `sandbox` flag in `webPreferences` guessing that it might help and it did.

If I'm not wrong @nornagon was going to start the electron in dev mode with sandbox disabled by default (could not find the related issue) which is actually happening in v8-beta1 as described above (with help of adding `--no-sandbox` arg). So I guess @nornagon could shed some light on this issue.
